pkg/userutils/userutil_linux.go

```diff
@@ -0,0 +1,145 @@
+// +build linux
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package userutils
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	gouser "os/user"
+	"strconv"
+	"syscall"
+
+	"github.com/aws-observability/aws-otel-collector/pkg/logger"
+	"github.com/opencontainers/runc/libcontainer/system"
+	"github.com/opencontainers/runc/libcontainer/user"
+	"golang.org/x/sys/unix"
+)
+
+var defaultUser = "aoc"
+
+// switchUser swithes the current process to new user
+func switchUser(execUser *user.ExecUser) error {
+	if err := unix.Setgroups(execUser.Sgids); err != nil {
+		log.Printf("E! Failed to set groups: %v", err)
+		return err
+	}
+
+	if err := system.Setgid(execUser.Gid); err != nil {
+		log.Printf("E! Failed to set gid: %v", err)
+		return err
+	}
+
+	if err := system.Setuid(execUser.Uid); err != nil {
+		log.Printf("E! Failed to set uid: %v", err)
+		return err
+	}
+
+	if err := os.Setenv("HOME", execUser.Home); err != nil {
+		log.Printf("E! Failed to set HOME: %v", err)
+		return err
+	}
+	log.Printf("I! Set HOME: %v", execUser.Home)
+
+	return nil
+}
+
+// getRunAsExecUser returns the new user uid and gid
+func getRunAsExecUser(runasuser string) (*user.ExecUser, error) {
+	currExecUser := user.ExecUser{
+		Uid:  syscall.Getuid(),
+		Gid:  syscall.Getgid(),
+		Home: "/root",
+	}
+	newUser, err := user.GetExecUserPath(runasuser, &currExecUser, "/etc/passwd", "/etc/group")
+	if err != nil {
+		log.Printf("E! Failed to get newUser: %v", err)
+		return nil, err
+	}
+	return newUser, nil
+}
+
+// ChangeUser allow customers to run the collector in selected user
+// by default it ran as 'aoc' user but can be set by environment variable
```

Comment:
if you change aoc to some other user name, change the comment as well

Reply:
How to use customized user?

Reply:
added `func getCustomUser() string {}` ? but we iterate it later if read customized user from Env Var doesn't work well.